relays/ethereum/src/ethereum_exchange_submit.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Submitting Ethereum -> Substrate exchange transactions.
+
+use crate::ethereum_client::{EthereumConnectionParams, EthereumRpcClient, EthereumSigningParams};
+use crate::ethereum_types::{CallRequest, U256};
+use crate::rpc::EthereumRpc;
+
+use bridge_node_runtime::exchange::LOCK_FUNDS_ADDRESS;
+use hex_literal::hex;
+use sp_bridge_eth_poa::{
+	signatures::{SecretKey, SignTransaction},
+	UnsignedTransaction,
+};
+
+/// Ethereum exchange transaction params.
+#[derive(Debug)]
+pub struct EthereumExchangeSubmitParams {
+	/// Ethereum connection params.
+	pub eth: EthereumConnectionParams,
+	/// Ethereum signing params.
+	pub eth_sign: EthereumSigningParams,
+	/// Ethereum signer nonce.
+	pub eth_nonce: Option<U256>,
+	/// Amount of Ethereum tokens to lock.
+	pub eth_amount: U256,
+	/// Funds recipient on Substrate side.
+	pub sub_recipient: [u8; 32],
+}
+
+impl Default for EthereumExchangeSubmitParams {
+	fn default() -> Self {
+		EthereumExchangeSubmitParams {
+			eth: Default::default(),
+			eth_sign: Default::default(),
+			eth_nonce: None,
+			eth_amount: 1_000_000_000_000_000_000_u64.into(), // 1 ETH
+			sub_recipient: hex!("1cbd2d43530a44705ad088af313e18f80b53ef16b36177cd4b77b846f2a5f07c"), // ferdie
```

Comment:
Some pre-history - since the very beginning, all relay CLI options had default values that were pointing to: (1) some well-known addresses in OE development chain at PoA side (i.e. to 0x00a329c0648769a73afac7f9381e08fb43dbea72) and (2) some well-known addresses in our Substrate runtime (i.e. Alice, Bob, ...). So now we have default values for everything, including contract code, contract address (which depends on eth-signer and its nonce), sub-signers, ... The idea was that you could execute some relay subcommand  with both OE and Substrate nodes active and everything will work by default. 

I agree that these times it isn't actual anymore, because we (even myself) no longer use OE dev chain. But I'd prefer to leave all cli options consistent, than to make some work in has-defaults way and some in no-defaults way. I suggest to address this altogether in #162 by requiring values for all parameters + introducing configuration files (OE-like) as replacement for default values.

Feel free to insist on your change though - I'll add `required` to `cli.yml`. But I feel it is wrong way to fix it now.

Reply:
I do agree that adding `required` is mainly just a bandaid fix, but I think in this case we should favour safety over consistency

Reply:
IMHO this should be reworked as part of #162 I'd rather keep it consistent for now, maybe we will come back to `--dev` testing :)

Reply:
ok, then I'm merging